### PR TITLE
feat: Astra compatibility with Gutenberg block patterns

### PR DESCRIPTION
**Pull request by imnavanath**
### Description
- https://trello.com/c/8HMa6B4q/233-gutenberg-block-pattern-compatibility
- Compatibility with Buttons to buttons used in GB block patterns

### Types of changes
- Added backward compatibility for old users
- For new it start directly

### How has this been tested?
- For old setup
- Filter to use - `add_filter( 'astra_gutenberg_patterns_compatibility', '__return_true' );`

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards
- [x] My code has proper inline documentation
- [x] I've included any necessary tests 
- [x] I've included developer documentation
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -15,7 +15,7 @@
 /**
  * Define Constants
  */
-define( 'ASTRA_THEME_VERSION', '2.6.2' );
+define( 'ASTRA_THEME_VERSION', '2.6.3' );
```
Comment (premanshup) [inline functions.php:18]: @imnavanath Please pull latest `next-release` in this and change the backward comp version to 3.0.0 as well.

File: inc/core/class-gutenberg-editor-css.php [hunk 2]
```diff
@@ -426,6 +430,81 @@ public static function get_css() {
...
+						'padding-top'         => 'calc(15px - ' . (int) $theme_btn_top_border . 'px)',
+						'padding-right'       => 'calc(30px - ' . (int) $theme_btn_right_border . 'px)',
+						'padding-bottom'      => 'calc(15px - ' . (int) $theme_btn_bottom_border . 'px)',
+						'padding-left'        => 'calc(30px - ' . (int) $theme_btn_left_border . 'px)',
+					),
+					'.wp-block-button.is-style-outline > .wp-block-button__link:not(.has-text-color), .wp-block-button.wp-block-button__link.is-style-outline:not(.has-text-color)' => array(
+						'color' => empty( $btn_border_color ) ? esc_attr( $btn_bg_color ) : esc_attr( $btn_border_color ),
+					),
+					'.wp-block-button.is-style-outline .wp-block-button__link:hover, .wp-block-button.is-style-outline .wp-block-button__link:focus' => array(
+						'color' => esc_attr( $btn_text_hover_color ) . ' !important',
```
Comment (premanshup) [inline inc/core/class-gutenberg-editor-css.php:466]: @imnavanath Can we remove `!important` if possible here?

**Review by premanshup — CHANGES_REQUESTED** [review]

**Review by shubhw12 — APPROVED** [review]

File: functions.php (hunk 1 above)
Comment (imnavanath) [reply functions.php]: Updated to version `3.1.0` for both theme constant & backward function.

File: inc/core/class-gutenberg-editor-css.php (hunk 2 above)
Comment (imnavanath) [reply inc/core/class-gutenberg-editor-css.php:466]: I tried this, but the CSS applies inline through GB block. That's why I've added important here.

**Review by premanshup — APPROVED** [review]

**Review by vrundakansara — APPROVED** [review]

**Pull request merged**